src/core/library.c

```diff
@@ -663,6 +663,54 @@ QuicLibraryGetGlobalParam(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_GLOBAL_PERF_COUNTERS: {
+
+        if (*BufferLength < sizeof(uint64_t)) {
+            *BufferLength = sizeof(uint64_t) * QUIC_PERF_COUNTER_MAX;
+            Status = QUIC_STATUS_BUFFER_TOO_SMALL;
+            break;
+        }
+
+        if (Buffer == NULL) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        uint64_t MergedPerfCounters[QUIC_PERF_COUNTER_MAX];
+        memcpy(MergedPerfCounters, MsQuicLib.PerProc[0].PerfCounters, sizeof(MergedPerfCounters));
+
+        for (int ProcIndex = 1; ProcIndex < MsQuicLib.PartitionCount; ++ProcIndex) {
+            for (int CounterIndex = 0; CounterIndex < QUIC_PERF_COUNTER_MAX; ++CounterIndex) {
+                switch (CounterIndex) {
+                    case ConnectionsActive:
+                        MergedPerfCounters[CounterIndex] =
+                            ((int64_t)MergedPerfCounters[CounterIndex]) +
+                            (int64_t)MsQuicLib.PerProc[ProcIndex].PerfCounters[CounterIndex];
+                        break;
+                    default:
+                        MergedPerfCounters[CounterIndex] +=
+                            MsQuicLib.PerProc[ProcIndex].PerfCounters[CounterIndex];
+                        break;
+                }
+            }
+        }
+
+        QUIC_DBG_ASSERT((int64_t)MergedPerfCounters[ConnectionsActive] >= 0);
```

Comment:
Dont' assert this. It's theoretically possible, depending on timings that a value could be less than zero. Instead, if it is less than zero, set it to zero.